### DISCUSSION TITLE
静态链接libEGL.dll，libGLESv2.dll（或裁剪掉它们）

### DISCUSSION THREAD

**Discussion by xland** [Tencent/tgfx#397] [General] [answered]
我编译了Hello2D.exe，
发现Hello2D.exe运行依赖libEGL.dll，libGLESv2.dll
请问有办法静态链接这个库吗？

另外，这两个库加起来有16M，有没有裁剪掉这两个库的可能性？

**Comment by domchen** [comment, marked as answer]
tgfx是跨所有平台的，下次提问要说一下具体是什么平台的。这两个库是ANGLE的，在Windows上模拟提供OpenGL环境。ANGLE没法静态链接，因为它内部就有接口需要依赖动态的函数指针。这两个也没法裁剪，Windows平台多少都得带点DLL的，16M的依赖对桌面程序包体也并不算大。或者可以等后面我们在Windows平台支持原生的WGL后端时，可以选择那个，应该可以更小一点。

**Reply by xland** [reply]
多谢回复！期待新版本~

**Comment by xland** [comment]
今天刚编译了一下最新代码，发现已经有纯OpenGL后端的版本了，可以不再依赖ANGLE了

**Reply by domchen** [reply]
是的，现在有原生的WGL支持了。
